packages/gateway-trpc/src/modules/event/event-router.ts

```diff
@@ -66,53 +68,34 @@ export const eventRouter = t.router({
       return event
     }),
 
-  // TODO: N+1 query, eventHostingGroupService and eventService should probably be merged
   all: publicProcedure
-    .input(
-      z
-        .object({
-          page: PaginateInputSchema,
-          filter: EventFilterSchema.optional(),
-        })
-        .optional()
-    )
+    .input(z.object({ page: PaginateInputSchema, filter: EventFilterSchema }).optional())
+    .output(z.array(EventDetailSchema))
     .query(async ({ input, ctx }) => {
       const events = await ctx.eventService.getEvents(input?.page, input?.filter)
       const groups = events.map(async (e) => ctx.eventHostingGroupService.getHostingGroupsForEvent(e.id))
       const interestGroups = events.map(async (e) => ctx.interestGroupService.getAllByEventId(e.id))
       const companies = events.map(async (e) => ctx.companyEventService.getCompaniesByEventId(e.id))
+      const attendances = events.map(async (e) =>
+        e.attendanceId ? ctx.attendanceService.getById(e.attendanceId) : null
+      )
 
       const groupResults = await Promise.all(groups)
       const interestGroupResults = await Promise.all(interestGroups)
       const companyResults = await Promise.all(companies)
+      const attendanceResults = await Promise.all(attendances)
 
-      return events.map((event, i) => ({
-        ...event,
-        groups: groupResults[i],
-        interestGroups: interestGroupResults[i],
-        companies: companyResults[i],
-      }))
+      return events.map(
+        (event, i): EventDetail => ({
+          event,
```

Comment:
her er culprit'en for at dashboardet crasher @jotjern. Event-propertiesa blir lagt i .event istedenfor spread rett inn i objektet. https://github.com/dotkom/monoweb/blob/a8269fa6ea01e246deba27a304ec76eef0fb50a1/apps/dashboard/src/app/(dashboard)/event/page.tsx#L24-L58 forventer at propertiesa ligger rett i objektet.